src/contracts/logic/math.rs

```diff
@@ -0,0 +1,444 @@
+use decimal::*;
+
+use crate::math::liquidity::Liquidity;
+use crate::math::sqrt_price::sqrt_price::{calculate_sqrt_price, SqrtPrice};
+use crate::math::token_amount::TokenAmount;
+use crate::math::MAX_TICK;
+
+const PRICE_DENOMINATOR: u128 = 1_000_000_000_000_000_000_000_000;
```

Comment:
do not duplicate decimal consts. You can use:
`Price::one(), Price::from_integer(1), 10^Price::scale` etc